amundsen_application/static/js/components/TableDetail/index.tsx

```diff
@@ -165,6 +165,8 @@ export class TableDetail extends React.Component<
       innerContent = <ErrorMessage />;
     } else {
       const data = tableData;
+      const editText = data.source ? data.source.source_type : null;
```

Comment:
1. Let's have a more helpful piece of text like we do for dashboards:
![image](https://user-images.githubusercontent.com/1790900/86070739-43fce600-ba32-11ea-99ac-128260f234e0.png)

2. `editText` and `editUrl` are typed as options strings, so let's use empty strings instead of `null` for the default. If we ever enable a stricter type checking option in the Typescript code,  these lines would end up throwing errors